setup.py

```diff
@@ -1,26 +1,84 @@
+import os
+import subprocess
 from setuptools import setup, find_packages
 
 with open("README.md", "r") as fh:
     long_description = fh.read()
 
-setup_attrs = {
-    'name': 'torchani',
-    'description': 'PyTorch implementation of ANI',
-    'long_description': long_description,
-    'long_description_content_type': "text/markdown",
-    'url': 'https://github.com/aiqm/torchani',
-    'author': 'Xiang Gao',
-    'author_email': 'qasdfgtyuiop@gmail.com',
-    'license': 'MIT',
-    'packages': find_packages(),
-    'include_package_data': True,
-    'use_scm_version': True,
-    'setup_requires': ['setuptools_scm'],
-    'install_requires': [
+
+def cuda_extension():
+    import torch
+    from torch.utils.cpp_extension import CUDAExtension
+
+    nvcc_args = ["-gencode=arch=compute_50,code=sm_50", "-gencode=arch=compute_60,code=sm_60",
+                 "-gencode=arch=compute_61,code=sm_61", "-gencode=arch=compute_70,code=sm_70",
+                 "-Xptxas=-v", '--expt-extended-lambda', '-use_fast_math']
+    include_dirs = []
+    cuda_version = float(torch.version.cuda)
+    if cuda_version >= 10:
+        nvcc_args.append("-gencode=arch=compute_75,code=sm_75")
+    if cuda_version >= 11:
+        nvcc_args.append("-gencode=arch=compute_80,code=sm_80")
+    if cuda_version >= 11.1:
+        nvcc_args.append("-gencode=arch=compute_86,code=sm_86")
+    if cuda_version < 11:
+        # if no cub, download it to include dir from github
+        if not os.path.isdir('./include/cub'):
+            if not os.path.exists('./include'):
+                os.makedirs('include')
+            commands = """
+            echo "Downloading CUB library";
+            wget -q https://github.com/NVIDIA/cub/archive/main.zip;
```

Comment:
CUB can be installed with conda `conda install -c conda-forge cub`. Downloading the include files during the build will be an issue for the conda build because of licensing.

One way to support both installation is to modify the `if not os.path.isdir('./include/cub'):` logic and instead check for a CUB file in all the include paths. Maybe you can use `torch.utils.cpp_extension.include_paths(cuda=True)` and iterate over all the returned paths?

Reply:
I have updated `setup.py` as you suggest.

Reply:
Nice. Building the conda package will be much easier that way. Thanks.